packages/core/parcel-bundler/src/Pipeline.js

```diff
@@ -117,7 +116,9 @@ class Pipeline {
         type,
         value: asset.generated[type],
         // for scope hoisting, we need to post process all JS
-        final: !(type === 'js' && this.options.scopeHoist)
+        final: asset.hasOwnProperty('needsPipelineProcessing')
+          ? !asset.needsPipelineProcessing
```

Comment:
Why is this new property needed? Normally, if the `type` of the asset changes (e.g. from `.md` to `.html`), the pipeline will automatically pass it through to the next asset type (in this case `HTMLAsset`).

Reply:
Thanks for your comment!

> Normally, if the type of the asset changes (e.g. from .md to .html), the pipeline will automatically pass it through to the next asset type

Could you, please point out where that happens exactly? 

From what we discovered, the only way for an asset to undergo one more processing by another asset type is for this `final` property to be `false`.

https://github.com/parcel-bundler/parcel/blob/bdc044acfad84ba25bdb6631fbd3dd955424a227/packages/core/parcel-bundler/src/Pipeline.js#L55-L58

As you can see in the diff, the original code only allows `final` to be `false` for JS.

Maybe the right way to go would be to remove those 4 lines above altogether. 😱 But I suppose they are there for a reason.

Please, let me know if I've got it wrong.

Reply:
Ohhh, I've got it!